packages/ui/src/helpers/authenticator/utils.ts

```diff
@@ -91,6 +91,9 @@ export const defaultAuthHubHandler: AuthMachineHubHandler = async (
         send('TOKEN_REFRESH');
       }
       break;
+    case 'signInWithRedirect':
+      send('REFRESH');
+      break;
```

Comment:
Concern here is that we may have introduced a race condition somewhere else in the machine (@esauerbo has tested Vue and Angular with the current machine and had no issue) and that this might mask a deeper issue that will bite us somewhere else